lib/transports/rest.js

```diff
@@ -0,0 +1,386 @@
+'use strict'
+
+const crypto = require('crypto')
+const request = require('request')
+
+const { genAuthSig } = require('../util')
+
+const API_URL = 'https://api.bitfinex.com'
+
+/**
+ * Communicates with v1 of the Bitfinex HTTP API
+ */
+class RESTv1 {
+  constructor (opts = { apiKey: '', apiSecret: '', url: API_URL }) {
+    this._url = opts.url || API_URL
+    this._apiKey = opts.apiKey || ''
+    this._apiSecret = opts.apiSecret || ''
+    this._nonce = Date.now()
+    this._generateNonce = (typeof opts.nonceGenerator === 'function')
+        ? opts.nonceGenerator
+        : () => ++this._nonce
+  }
+
+  make_request (path, params, cb) {
+    if (!this._apiKey || !this._apiSecret) {
+      return cb(new Error('missing api key or secret'))
+    }
+
+    const payload = Object.assign({
+      request: `/v1/${path}`,
+      nonce: JSON.stringify(this._generateNonce())
+    }, params)
+
+    const payloadBase64 = new Buffer(JSON.stringify(payload)).toString('base64')
+    const { sig } = genAuthSig(this._apiSecret, payloadBase64)
+
+    return request({
+      url: `${this._url}/v1/${path}`,
+      method: 'POST',
+      timeout: 15000,
+      headers: {
+        'X-BFX-APIKEY': this._apiKey,
+        'X-BFX-PAYLOAD': payloadBase64,
+        'X-BFX-SIGNATURE': sig
+      }
+    }, (err, response, body) => {
+      if (err || (response.statusCode !== 200 && response.statusCode !== 400)) {
+        return cb(new Error(err != null ? err : response.statusCode))
```

Comment:
this is tricky because of the implicit bool conversion via `!=`

ok so we just get into this path if error is defined or if the status code does not fit. btw what is with HTTP `201`? Maybe test via regex: /^2../test(response.statusCode)

suggestion:

```js
if (err) return cb(err)
if (response.statusCode !== 200 && response.statusCode !== 400) return cb(new Error(response.statusCode))

// yada yada
```